feat: Adds `Navbar` component (#1720)
vtex/faststore: apps/site/pages/components/features/navigation.mdx

import { Navbar, UIProvider } from '@faststore/ui'
import { SectionItem, SectionList } from 'site/components/SectionItem'

import NavbarUsage from 'site/components/Navigation/NavbarUsage'
import NavbarLinksUsage from 'site/components/Navigation/NavbarLinksUsage'
import NavbarSliderUsage from 'site/components/Navigation/NavbarSliderUsage'

export const NavigationUsage = () => null

<header>

# Navigation

This feature enables customers to navigate among the store pages.

</header>

## Overview

> This feature is intended to group the main store's page header components. These components perform specific functions such as navigation among the store pages, navigation to the login flow and others.

Reviewer: ```suggestion
This feature aims to group the main store's page header components. These components perform specific functions such as navigation among the store pages, navigation to the login flow and others.
```

---

## Usage

> There is an available example of the usage of all the related components in the `<Navbar />` component on our [core](https://github.com/vtex/faststore/tree/main/packages/core/src/components/navigation/Navbar/Navbar.tsx) application.

Reviewer: ```suggestion
There is an available example of the usage of all the related components in the `<Navbar />` component on our [core](https://github.com/vtex/faststore/blob/main/packages/core/src/components/common/Navbar/Navbar.tsx) application.
```

Reviewer: actually, let's check this link later once again hahaha

Author: Yeah hahaha this one is the new link, so it won't work until the PR is merged

> The recommendation is to use the components together, keeping the `Navbar` as the wrapper of the others.

Reviewer: ```suggestion
We recommend the usage of the components together, keeping the `Navbar` as the wrapper of the others.
```
